internal/services/payment_from_submitter_service.go

```diff
@@ -85,21 +84,25 @@ func (s PaymentFromSubmitterService) syncTransaction(ctx context.Context, dbTx d
 	return nil
 }
 
-// syncPaymentsWithTransactions updates the status of the payments based on the status of the transactions.
-func (s PaymentFromSubmitterService) syncPaymentsWithTransactions(ctx context.Context, dbTx db.DBTransaction, transaction *txSubStore.Transaction) error {
+// syncPaymentsWithTransaction updates the status of the payments based on the status of the transactions.
+func (s PaymentFromSubmitterService) syncPaymentsWithTransaction(ctx context.Context, dbTx db.DBTransaction, transaction *txSubStore.Transaction) error {
 	payments, err := s.sdpModels.Payment.GetByIDs(ctx, dbTx, []string{transaction.ExternalID})
 	if err != nil {
 		return fmt.Errorf("getting payments by IDs: %w", err)
 	}
 
-	if len(payments) == 0 {
+	if len(payments) != 1 {
 		return fmt.Errorf("no payment found for transaction ID %s", transaction.ID)
```

Comment:
Can you update the error message with helpful information? You can say that a one payment was expected but the query returned `len(payments)` instead.